cli/js/web/storage.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import * as domTypes from "./dom_types.d.ts";
+
+const data = Symbol("internal Storage data");
+
+class Storage implements domTypes.Storage {
+  private [data]: Map<string, string> = new Map();
+
+  get length(): number {
+    return this[data].size;
+  }
+  key(index: number): string | null {
+    if (index >= this[data].size) {
+      return null;
+    } else {
+      return Array.from(this[data].keys())[index];
+    }
+  }
+  getItem(keyName: string): string | null {
+    return this[data].get(keyName) || null;
+  }
+  setItem(keyName: string, keyValue: string): void {
+    this[data].set(keyName, keyValue);
+  }
+  removeItem(keyName: string): void {
+    this[data].delete(keyName);
+  }
+  clear(): void {
+    this[data].clear();
+  }
+}
+
+const storageHandler = {
+  deleteProperty(target: domTypes.Storage, key: string): boolean {
+    target.removeItem(key);
+    return true;
+  },
+  has(target: domTypes.Storage, key: string): boolean {
+    return target.getItem(key) !== null;
+  },
+  get(target: domTypes.Storage, key: string): string | null | number | void {
+    if ("undefined" !== typeof target[key]) {
+      // @ts-ignore
+      return Reflect.get(...arguments); // eslint-disable-line prefer-rest-params
+    } else {
+      return target.getItem(key);
+    }
+  },
+  set(target: domTypes.Storage, key: string, value: string): boolean {
+    try {
+      target.setItem(key, value);
+      return true;
+    } catch {
+      return false;
+    }
+  },
+};
+
+export const sessionStorage = new Proxy(
```

Comment:
Why do we need to use a proxy?

Reply:
To implement the Object-like access to items:

```js
// These three lines all set the (same) colorSetting entry
localStorage.colorSetting = '#a4509b';
localStorage['colorSetting'] = '#a4509b';
localStorage.setItem('colorSetting', '#a4509b');
```

Reply:
ah, ok, yeah, a symbol is appropriate then... if you had just used `private` then you would never have been able to have a property named `data` then, right? 😄

Reply:
Yeah, that would have created some weird bugs 🙃